### PR TITLE
PYTHON-3920 - Migrate AWS Auth Tests to use AWS Secrets

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -553,39 +554,21 @@ functions:
...
 
-  "add aws auth variables to file":
+  "get aws auth secrets":
+    - command: ec2.assume_role
+      params:
+        role_arn: ${aws_test_secrets_role}
     - command: shell.exec
       type: test
       params:
+        add_expansions_to_env: true
```
Comment (blink1073) [inline .evergreen/config.yml:564]: These can be simplified to `include_expansions_in_env: ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN"]`.  I tried it out in https://github.com/mongodb/mongo-go-driver/pull/1365

File: .evergreen/config.yml [hunk 2]
```diff
@@ -664,29 +635,33 @@ functions:
...
-             exit 0
-          fi
-          # Write an empty prepare_mongodb_aws so no auth environment variables
-          # are set.
-          rm "${PROJECT_DIRECTORY}/prepare_mongodb_aws.sh" || true
-          PYTHON_BINARY=${PYTHON_BINARY} ASSERT_NO_URI_CREDS=true .evergreen/run-mongodb-aws-test.sh
+
+          cd -
+
+          # Write an empty prepare_aws_env so no auth environment variables are set.
```
Comment (blink1073) [inline .evergreen/config.yml:660]: We shouldn't need to do this anymore (perhaps something still needs to change in D-E-T?

File: .evergreen/config.yml [hunk 3]
```diff
@@ -695,60 +670,42 @@ functions:
...
         shell: "bash"
         script: |
-          ${PREPARE_SHELL}
           if [ "${skip_EC2_auth_test}" = "true" ]; then
              echo "This platform does not support the web identity auth test, skipping..."
              exit 0
           fi
-          set -ex
+
+          # Try to source exported AWS Secrets
```
Comment (blink1073) [inline .evergreen/config.yml:678]: We can probably move this logic into `run-mongodb-aws-test.sh` and pass in the test to run (`web-identity` in this case).

File: .evergreen/run-tests.sh [hunk 4]
```diff
@@ -69,6 +69,7 @@ if [ -n "$TEST_ENTERPRISE_AUTH" ]; then
         echo "Writing keytab"
         echo ${KEYTAB_BASE64} | base64 -d > ${PROJECT_DIRECTORY}/.evergreen/drivers.keytab
         echo "Running kinit"
+        python -c "import os;print(sorted([(k, v[:2]) for k, v in os.environ.items()]))"
```
Comment (blink1073) [inline .evergreen/run-tests.sh:72]: Remove?

**Review by blink1073 — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml (hunk 2 above)
Comment (blink1073) [reply .evergreen/config.yml:660]: Ah, I see how it is used now.  I think we can also absorb this logic into `run-mongodb-aws-test.sh`.  I do like the idea of adding the script in D-E-T as you've done.

**Comment by blink1073** [comment]
Nice, I like the direction this is heading.  Looks like you still need to convert the last few tests?

**Comment by NoahStapp** [comment]
> Nice, I like the direction this is heading. Looks like you still need to convert the last few tests?

Should all be converted and good to go now!

File: .evergreen/config.yml (hunk 1 above)
Comment (blink1073) [reply .evergreen/config.yml:565]: This particular instance still needs to be updated.  :)

Comment (NoahStapp) [reply .evergreen/config.yml:565]: for `get_aws_secrets` it fails without `expansions_to_env`, `test-atlas` has been updated to use `include_expansions_in_env`

Comment (blink1073) [reply .evergreen/config.yml:565]: Wait so there are things we're missing from the new vault?

Comment (NoahStapp) [reply .evergreen/config.yml:565]: `get_aws_secrets` is what fetches the secrets from the vault.

Comment (blink1073) [reply .evergreen/config.yml:565]: Right, okay, so now I'm confused, because that's what I'm doing in the go driver ticket.

**Comment by NoahStapp** [comment]
> Beautiful!

Are these failures expected? 

Screen Shot 2023-09-05 at 11 20 28 AM

**Comment by blink1073** [comment]
No, I something changed on those tests, I'll open a ticket.

**Comment by NoahStapp** [comment]
> No, I something changed on those tests, I'll open a ticket.

Is this good to merge then?

**Comment by blink1073** [comment]
> Is this good to merge then?

Yes, I opened https://jira.mongodb.org/browse/PYTHON-3951

**Pull request merged**